eos-updater-prepare-volume/eos-updater-prepare-volume

```diff
@@ -0,0 +1,228 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+# Copyright © 2017 Endless Mobile, Inc.
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+# Boston, MA 02111-1307, USA.
+
+import argparse
+import os
+import subprocess
+import sys
+
+import gi
+gi.require_version('Flatpak', '1.0')
+gi.require_version('GLib', '2.0')
+gi.require_version('OSTree', '1.0')
+from gi.repository import Flatpak, GLib, OSTree  # noqa
+
+
+class VolumePreparer:
+    EXIT_FAILED = 1
+    EXIT_INVALID_ARGUMENTS = 2
+    EXIT_RUN_AS_ROOT = 3
+
+    def __init__(self, volume_path, flatpak_refs, quiet=False, debug=False):
+        self.volume_path = volume_path
+        self.flatpak_refs = flatpak_refs
+        self.quiet = quiet
+        self.debug = debug
+
+        self.sysroot = OSTree.Sysroot.new_default()
+
+    def __run(self, cmd, use_shell=False):
```

Comment:
I don’t care about this code being reusable at the moment (it’s all fairly tightly tied to the command line tool’s needs), so would prefer to keep them private to the `VolumePreparer` class for the moment. They can always be factored out in future.

Reply:
Yeah its more of a philosophical point - I've seen a lot of python code where everything is all in one class where a class isn't (in my view) necessary. I think it may be a difference in stylistic preference though, so if you're happy with this design then I wouldn't object.